src/librustc_typeck/check/regionck.rs

```diff
@@ -219,7 +219,7 @@ impl<'a, 'tcx> RegionCtxt<'a, 'tcx> {
             body_id: initial_body_id,
             body_owner: subject,
             call_site_scope: None,
-            subject_def_id: subject,
+            subject_def_id: subject.to_def_id(),
```

Comment:
`subject_def_id` should probably be `LocalDefId` too.

Reply:
Fixed!